lib/memory.h

```diff
@@ -75,15 +75,20 @@ struct memgroup {
  *         but MGROUP_* aren't.
  */
 
+#ifdef __MACH__
+#define ATTRIBUTE_MGROUPS __attribute__((section("__DATA,mgroups")))
+#else
+#define ATTRIBUTE_MGROUPS __attribute__((section(".data.mgroups")))
+#endif
```

Comment:
ACK, but

* this belongs in `lib/compiler.h`
* please make it one macro with a parameterized section name

```sugg
#ifdef __MACH__
#define _DATA_SECTION(name) __attribute__((section("__DATA," name)))
#else
#define _DATA_SECTION(name) __attribute__((section(".data." name)))
#endif
```

(Note consecutive string literals in C are automatically concatenated, `"foo" "bar"` is the same as `"foobar"`.  This does work correctly for GCC attribute parameters.)

would be used like this:
```struct memgroup _mg_##mname _DATA_SECTION("mgroups")```

(we have some macros with the `__attribute__` included in the macro, and some without the `__attribute__` - I don't really care much either way on that...  maybe we should clean that up and make it consistent one way or another.)